### ISSUE TITLE
got stuck in this rule convert_to_ref, convert_reference did not work

### ISSUE THREAD

**Issue by Jamesped** [cbg-ethz/V-pipe#70]
Job counts:
	count	jobs
	1	convert_to_ref
	1

        convert_reference -t gi|1142969405|gb|KY272010.1| -m references/ALL_aln_ambig.fasta -i samples/HFMD/71147/alignments/full_aln.bam -o samples/HFMD/71147/alignments/REF_aln.bam > samples/HFMD/71147/alignments/convert_to_ref.out.log 2> >(tee samples/HFMD/71147/alignments/convert_to_ref.err.log >&2)

/bin/bash: 1142969405: command not found
/bin/bash: gb: command not found
/bin/bash: KY272010.1: command not found
/bin/bash: -m: command not found
/bin/bash: 1142969405: command not found
/bin/bash: gb: command not found
/bin/bash: KY272010.1: command not found
/bin/bash: -m: command not found
/bin/bash: 1142969405: command not found
/bin/bash: gb: command not found
/bin/bash: KY272010.1: command not found
/bin/bash: 1142969405: command not found
/bin/bash: gb: command not found
/bin/bash: KY272010.1: command not found
/bin/bash: -m: command not found
/bin/bash: -m: command not found
usage: convert_reference [-h] -t TO [-v] -m input -i input [-o output] [-p]
                         [-X] [-H]
convert_reference: error: the following arguments are required: -m, -i
usage: convert_reference [-h] -t TO [-v] -m input -i input [-o output] [-p]
                         [-X] [-H]
convert_reference: error: the following arguments are required: -m, -i
usage: convert_reference [-h] -t TO [-v] -m input -i input [-o output] [-p]
                         [-X] [-H]
convert_reference: error: the following arguments are required: -m, -i
usage: convert_reference [-h] -t TO [-v] -m input -i input [-o output] [-p]
                         [-X] [-H]
convert_reference: error: the following arguments are required: -m, -i
[Fri Oct 30 07:39:20 2020]
Error in rule convert_to_ref:
    jobid: 0
    output: samples/HFMD/71157/alignments/REF_aln.bam
    log: samples/HFMD/71157/alignments/convert_to_ref.out.log, samples/HFMD/71157/alignments/convert_to_ref.err.log (check log file(s) for error message)
    conda-env: /Volumes/AKiTiO_duo3/CoxA10/trimmed/match_to_virus_genome/v-pipe-working-dir/.snakemake/conda/77a26a2e
    shell:

        convert_reference -t gi|1142969405|gb|KY272010.1| -m references/ALL_aln_ambig.fasta -i samples/HFMD/71157/alignments/full_aln.bam -o samples/HFMD/71157/alignments/REF_aln.bam > samples/HFMD/71157/alignments/convert_to_ref.out.log 2> >(tee samples/HFMD/71157/alignments/convert_to_ref.err.log >&2)

        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Exiting because a job execution failed. Look above for error message

**Comment by Jamesped** [comment]
I think the problem is the first line of my reference fasta file. It was fixed with no space in this line.

**Comment by DrYak** [comment]
Yes, indeed the problem is triggered by the presence of the pipe symbol ("|") in the FASTA file.
This confuses bash which think these are all separate commands that need to filter each-other's output in a chain.

As a workaround, avoiding using non alphanumeric characters should indeed solve the problem.
